community/dbms/src/main/java/org/neo4j/commandline/dbms/Util.java

```diff
@@ -31,6 +33,28 @@
 
 public class Util
 {
+    public static Path canonicalPath( Path path ) throws IllegalArgumentException
+    {
+        return canonicalPath( path.toFile() );
+    }
+
+    public static Path canonicalPath( String path ) throws IllegalArgumentException
+    {
+        return canonicalPath( new File( path ) );
+    }
+
+    public static Path canonicalPath( File file ) throws IllegalArgumentException
+    {
+        try
+        {
+            return Paths.get( file.getCanonicalPath() );
+        }
+        catch ( IOException e )
+        {
+            throw new IllegalArgumentException( "Unable to parse path: " + file );
```

Comment:
Please include the original exception… the kinds of failures that can pop out of here are platform specific, and we don't want to miss the root cause when customers bring us stack traces asking for an explanation.